src/test/java/com/ge/predix/uaa/token/lib/ZacTokenServiceTest.java

```diff
@@ -57,45 +57,62 @@
     public void testLoadAuthentication() {
         // testing when zone id is not null
         String zoneUserScope = SERVICEID + ".zones." + ZONE + ".user";
-        assertAuthentication(loadAuthentication(PREDIX_ZONE_HEADER_NAME, BASE_DOMAIN, ZONE, zoneUserScope), ZONE);
+        assertAuthentication(loadAuthentication(PREDIX_ZONE_HEADER_NAME, BASE_DOMAIN, ZONE, zoneUserScope, true), ZONE);
     }
 
+    @SuppressWarnings("unchecked")
     @Test(expectedExceptions = IllegalStateException.class)
     public void testLoadAuthenticationNoHeaderOrBaseDomain() throws Exception {
         // testing when zone id is not null
-        ZacTokenService zacTokenServices = configureZacTokenService("", null, "");
+        ZacTokenService zacTokenServices = configureZacTokenService("", null, "", true, Collections.EMPTY_LIST,
+                Collections.EMPTY_LIST, null);
         zacTokenServices.afterPropertiesSet();
     }
 
+    @Test(expectedExceptions = InvalidRequestException.class)
+    public void testLoadAuthenticationNoSubdomains() throws Exception {
+        // testing when zone id is not null
+        String zoneUserScope = "scope.not.used";
+        loadAuthenticationWithZoneAsSubdomain(PREDIX_ZONE_HEADER_NAME, BASE_DOMAIN, ZONE, false, zoneUserScope,
+                "/test/resource", Arrays.asList("/zone/**"));
+    }
+
+    public void testLoadAuthenticationWithSubdomains() throws Exception {
+        // testing when zone id is not null
```

Comment:
testLoadAuthenticationWithSubdomains duplicates testLoadAuthentication.  The only difference is how the zone is passed in which is covered by tests for HttpServletRequestUtil.getZoneName